whylog_vim/gui/windows.py

```diff
@@ -0,0 +1,95 @@
+import vim
+
+from whylog_vim.consts import WindowTypes
+from whylog_vim.gui.vim_commander import VimCommander
+
+
+class WindowContext(object):
+
+    def __enter__(self):
+        VimCommander.set_modifiable()
+        return VimCommander.get_buffor()
+
+    def __exit__(self, type_, value, traceback):
+        VimCommander.set_nomodifible()
+
+
+class Window(object):
+
+    def __init__(self, name, content, modifiable, size=None):
+        content = content or ''
+        self.context = WindowContext()
+        if size is None:
+            VimCommander.open_file_at_window(name)
+        else:
+            VimCommander.split_window(name)
+            VimCommander.resize(size)
+        VimCommander.set_nowritable()
+        self.set_output(content)
+        self.set_modifiable(modifiable)
+        self.name = name
+        VimCommander.set_file_type()
```

Comment:
it looks suspicious when you call set_file_type with no type as param. Please rename set_file_type.